src/allotropy/parsers/thermo_skanit/thermo_skanit_structure.py

```diff
@@ -0,0 +1,277 @@
+        )
+
+        meas_groups = []
+        for index, row in abs_df.iterrows():
+            well_letter = row.iloc[0]
+            for col_name, value in list(row.items())[1:]:
+                col_index = abs_df.columns.get_loc(col_name)
+                if isinstance(col_index, tuple):
+                    col_index = col_index[0]
+                if isinstance(index, int) and isinstance(col_index, int):
+                    well_name = str(name_df.iloc[index, col_index])
+                else:
+                    msg = f"Unable to identify well name for row {index} and column {col_index}"
+                    raise AllotropyParserError(msg)
+                abs_value = value
+                abs_well = AbsorbanceDataWell.create(
+                    well_location=well_letter + str(col_index),
+                    abs_value=abs_value,
+                    sample_name=well_name,
+                    detector_wavelength=wavelength,
+                )
+                meas_groups.append(
+                    MeasurementGroup(
+                        measurements=[abs_well],
+                        plate_well_count=plate_well_count,
+                        measurement_time=exec_time,
+                        experimental_data_identifier=session_name,
+                    )
+                )
+        return meas_groups
+
+    @staticmethod
+    def get_plate_well_count(layout_definitions_df: pd.DataFrame) -> int:
+        # Combine all non-empty values in the row where "Plate template" is found
+        plate_template_row = layout_definitions_df[
+            layout_definitions_df.iloc[:, 0].str.contains(
+                "Plate template", case=False, na=False
+            )
+        ]
+        plate_template_value = (
+            plate_template_row.iloc[0].dropna().str.cat(sep=" ")
+            if not plate_template_row.empty
+            else None
+        )
+        msg = (
+            "Unable to identify plate well count from Layout definitions tab. "
+            "Expected to find in Plate template description."
+        )
+        if plate_template_value is None:
+            raise AllotropyParserError(msg)
+        else:
+            # Extract the integer from the combined string
+            search_string = re.search(r"\d+", plate_template_value)
+            if search_string:
+                plate_template_number = int(search_string.group())
+                return plate_template_number
+            else:
+                raise AllotropyParserError(msg)
+
+    @staticmethod
+    def identify_abs_and_sample_dfs(
+        absorbance_sheet_df: pd.DataFrame,
+    ) -> tuple[pd.DataFrame, pd.DataFrame, float]:
+        # Initialize variables
+        start_reading_abs = False
+        start_reading_sample = False
+        data_between_abs_and_blank = []
+        data_between_sample_and_blank = []
+
+        # Iterate through each row
+        for _index, row in absorbance_sheet_df.iterrows():
```

Comment:
After cleaning the dataframes, this can get a little simpler: 

```
        # Iterate through each row
        for _, row in absorbance_sheet_df.iterrows():
            if "Wavelength" in row.iloc[0]:
                match = re.search(r"Wavelength:\s*(\d{3})\s*nm", row.iloc[0])
                if match:
                    wavelength = float(match.group(1))
                else:
                    msg = "Unable to identify Wavelength (nm) from Absorbance tab"
                    raise AllotropyParserError(msg)
            if "Abs" in row.values.astype(str):  # Check if 'abs' is in the row
                start_reading_abs = True
            if "Sample" in row.values.astype(str):  # Check if 'sample' is in the row
                start_reading_sample = True
            if "Autoloading" in row.iloc[0]:
                break

            if start_reading_sample:
                data_between_sample_and_blank.append(row)
            elif start_reading_abs:
                data_between_abs_and_blank.append(row)
```